presto-main/src/main/java/com/facebook/presto/operator/PageSourceOperator.java

```diff
@@ -71,6 +73,12 @@ public boolean isFinished()
     }
 
     @Override
+    public ListenableFuture<?> isBlocked()
+    {
+        return toListenableFuture(pageSource.isBlocked());
```

Comment:
I think if we make the following change we can avoid some of the futures overhead.
```java
CompletableFuture<?> pageSourceBlocked = pageSource.isBlocked();
return pageSourceBlocked.isDone() ? NOT_BLOCKED : toListenableFuture(pageSourceBlocked);